programs/github

```diff
@@ -21,13 +21,48 @@ function progressbar(f)
 	return left .. right
 end
 
+-- get flag arguments
+local acceptedFlags = {
+	'-b' = 'branch',
+	'--branch' = 'branch',
+	'-t' = 'tag',
+	'--tag' = 'tag',
+	'-a' = 'auth',
+	'--auth' = 'auth'
+}
+
+local findFlag = function(arg) return acceptedFlags[arg] end
+
+local passed = {...}
+local args = {}
+local flags = {}
+local flagValue = ''
+for k, arg in pairs(passed) do
+	local flagName = findFlag(arg)
+	if flagName and arg ~= flagValue then
+		flags[flagName] = passed[k + 1]
+		flagValue = flags[flagName]
+	elseif arg ~= action and arg ~= flagValue then
```

Comment:
Damn it! Lol... Suggestion for fix?

On Thu, Aug 25, 2016, 17:44 Eric Wieser <notifications@github.com> wrote:

> In programs/github
> <https://github.com/eric-wieser/computercraft-github/pull/11#discussion_r76338628>
> :
>
> > +	'-a' = 'auth',
> > +	'--auth' = 'auth'
> > +}
> > +
> > +local findFlag = function(arg) return acceptedFlags[arg] end
> > +
> > +local passed = {...}
> > +local args = {}
> > +local flags = {}
> > +local flagValue = ''
> > +for k, arg in pairs(passed) do
> > +	local flagName = findFlag(arg)
> > +	if flagName and arg ~= flagValue then
> > +		flags[flagName] = passed[k + 1]
> > +		flagValue = flags[flagName]
> > +	elseif arg ~= action and arg ~= flagValue then
>
> This fails on github clone -b mybranch mybranch
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eric-wieser/computercraft-github/pull/11/files/03a2b915ba2b0e7d4c14ccde85bfa009059c75fe#r76338628>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA41FniJrdZ8e0wQWnq49pI4CJQDN_Jzks5qjhrcgaJpZM4JsvBA>
> .
>